workflows: e2e: bump Cilium CLI to v0.14.2

jibi opened the pull request:
which includes a new basic egress gateway test.

Bumping required also some extra logic to install static routes for
external CIDRs and figure out the correct external node IPs value to
pass to the --external-from-cidrs connectivity test parameters

jibi commented:
CI is green: https://github.com/cilium/cilium/actions/runs/4916343207/jobs/8779978281

jibi commented:
> Is it expected that we are now updating the CLI in workflows piecemeal?

not sure what you mean here

jibi commented:
> > > Is it expected that we are now updating the CLI in workflows piecemeal?
> > 
> > 
> > not sure what you mean here
> 
> Why aren't we updating the CLI in all workflows?

ah, right, because this PR started with just the e2e suite, but I can surely update all the workflows

brb commented:
This PR is adding an additional coverage to the e2e tests (fromCIDR, EGW tests). To avoid slipping unexpected blockers again, I think it's fine to let the renovate to eventually pick other workflows in separate PR, and meanwhile merge this PR.

pchaigno commented:
> This PR is adding an additional coverage to the e2e tests (fromCIDR, EGW tests). To avoid slipping unexpected blockers again, I think it's fine to let the renovate to eventually pick other workflows in separate PR, and meanwhile merge this PR.

I'm not against merging that without waiting, but let's note that Renovate is unable to test the workflow changes. So we can't rely on Renovate to update the CLI in workflows (a change that clearly needs to be tested).

brlbil reviewed (approved):
The consensus seems to be it is OK to merge only this. LGTM.

jibi commented:
I'll fix the conflict and update the v1.13 workflow and then we can merge this :+1: 

After that (in a separate PR to avoid additional blockers) I'll update all the remaining workflows since I'm already on it, as they'll also need the additional bash logic to set `EXTERNAL_NODE_IPS_PARAM` and the person approving the renovate PR will probably not have context on that

jibi committed:
workflows: e2e: bump Cilium CLI to v0.14.2

which includes a new basic egress gateway test.

Bumping required also some extra logic to install static routes for
external CIDRs and figure out the correct external node IPs value to
pass to the --external-from-cidrs connectivity test parameters

Signed-off-by: Gilberto Bertin <jibi@cilium.io>

jibi commented:
green run of the workflow: https://github.com/cilium/cilium/actions/runs/4923287537/jobs/8794943496?pr=25194

force pushed to get rid of https://github.com/cilium/cilium/compare/b55e99ebc4195295f200b14fef0a1df81cdffc6f..c1a4f698bfe79a78ddba40da46268e30fc6879da and marking ready to merge
